unit_testing/test_labels.py

```diff
@@ -139,11 +139,30 @@ def test_compute_mean_squared_error():
     for x, y, z, _ in src.getNonZeroCoordinates():
         if z < 5:
             ref.data[x, y, z+2] = src.data[x, y, z]
-
     mse = sct_labels.compute_mean_squared_error(src, ref)
     assert mse == 1.1547005383792515
 
 
+def test_compute_mean_squared_error_warning(caplog):
+    src = fake_3dimage_sct()
+    ref = src.copy()
+
+    src.data = np.where(src.data == ref.data[0, 0, 0], 1500, ref.data)
+
+    sct_labels.compute_mean_squared_error(src, ref)
+    # Cannot use f-string in assert, I needed to create a variable before
+    string_form = f'Label mismatch: Labels [{ref.data[0,0,0]}] present in reference image but missing from input image.'
+    assert 'Label mismatch: Labels [1500.0] present in input image but missing from reference image.' in caplog.text
+    assert string_form in caplog.text
+
+
+def test_compute_mean_squared_error_no_warning(caplog):
+    src = fake_3dimage_sct()
+    ref = src.copy()
+    sct_labels.compute_mean_squared_error(src, ref)
+    assert 'WARNING' not in caplog.text
```

Comment:
This test will fail for any warning, even unrelated warnings (e.g. `DeprecationWarnings` from dependencies). I think that might make the test too broad to be useful, when we're interested in specifically the `Label mismatch` warning. 

(Similarly, the test names are a little too broad... it might be good to give them slightly more specific names, like `test_compute_mse_no_label_warning` and `test_compute_mse_label_warning`)

Reply:
Got it I will make it more specific :)

Reply:
done :)